tests/test_modeling_tf_common.py

```diff
@@ -399,6 +408,58 @@ def test_pt_tf_model_equivalence(self):
             max_diff = np.amax(np.abs(tf_hidden_states - pt_hidden_states))
             self.assertLessEqual(max_diff, 4e-2)
 
+            has_labels = any(
+                x in tf_inputs_dict_maybe_with_labels for x in ["labels", "next_sentence_label", "start_positions"]
+            )
+            if has_labels:
+
+                with torch.no_grad():
+                    pto = pt_model(**pt_inputs_dict_maybe_with_labels)
+                tfo = tf_model(tf_inputs_dict_maybe_with_labels, training=False)
+
+                # Some models' output class don't have `loss` attribute despite `labels` is used.
+                tf_loss = getattr(tfo, "loss", None)
+                pt_loss = getattr(pto, "loss", None)
+
+                # Some models require extra condition to return loss. For example, `BertForPreTraining` requires both
+                # `labels` and `next_sentence_label`.
+                # Moreover, some PT models return loss while the corresponding TF/Flax models don't.
+                if tf_loss is not None and pt_loss is not None:
+
+                    tf_loss = tf.math.reduce_mean(tf_loss).numpy()
+                    pt_loss = pt_loss.numpy()
+
+                    tf_nans = np.copy(np.isnan(tf_loss))
+                    pt_nans = np.copy(np.isnan(pt_loss))
+                    # the 2 losses need to be both nan or both not nan
+                    # (`TapasForQuestionAnswering` gives nan loss here)
+                    self.assertEqual(tf_nans, pt_nans)
+
+                    if not tf_nans:
+                        max_diff = np.amax(np.abs(tf_loss - pt_loss))
+                        # `TFFunnelForTokenClassification` (and potentially other TF token classification models) give
+                        # large difference (up to 0.1x). PR #15294 addresses this issue.
+                        # There is also an inconsistency between PT/TF `XLNetLMHeadModel`.
+                        # Before these issues are fixed & merged, set a higher threshold here to pass the test.
```

Comment:
For this reason, I think we should merge this PR last (in your batch of PRs). That way, we can set a low comparison threshold :)

Reply:
The failed tests seem irrelevant to this PR. But we can wait a bit - I will rebase in the next 2 days to see if everything will be OK.